### PR TITLE
No sshtunnel module

### PR DESCRIPTION
**Pull request by klanclos**
Removes dependency on the sshtunnel module.

### CODE REVIEW

**Review by joshjriley — APPROVED** [review]
Tested and changes look good to me.

**Comment by joshwalawender** [comment]
I'd like a bit of a discussion of the pros and cons of removing this.  Having the dependency makes installation harder yes, but in general having calls to the shell to run things strikes me as more fragile.  I've always had the goal in code of not depending on the shell unless absolutely necessary because it is not under our control.  At least with the `sshtunnel` package it is under our control in that we work in the KRO conda environment, so there should be no unexpected upgrade or changes in API (we can fix the version).  When calling out to the shell, that's not under our control.

**Comment by klanclos** [comment]
Having to carry around an import that's not part of the standard install is going to be more fragile. The ssh command line options have been in-place for more than a decade, probably more like two decades; they aren't going to change under our feet without a lot of advance warning. And if they did, the sshtunnel module would probably stop working too!

File: keck_vnc_launcher.py [hunk 1]
```diff
@@ -512,30 +515,35 @@ def open_ssh_tunnel(self, server, username, password, ssh_pkey, remote_port,
...
-
-            #if success, keep track of ssh threads and ports in use
-#             self.ssh_threads.append(thread)
-            self.ports_in_use[local_port] = [address_and_port, session_name,
-                                             thread]
-            return local_port
+        forwarding = f"{local_port}:localhost:{remote_port}"
+        command = ['ssh', '-l', username, '-L', forwarding, '-N', '-T', server]
+
+        if ssh_pkey is not None:
```
Comment (joshwalawender) [inline keck_vnc_launcher.py:524]: Can you add comment to explain this block of code?

File: keck_vnc_launcher.py [hunk 2]
```diff
@@ -512,30 +515,35 @@ def open_ssh_tunnel(self, server, username, password, ssh_pkey, remote_port,
...
+            command.append('-i')
+            command.append(ssh_pkey)
+
+        self.log.debug('ssh command: ' + ' '.join (command))
+        process = subprocess.Popen(command)
+
+        if process.poll() is not None:
```
Comment (joshwalawender) [inline keck_vnc_launcher.py:536]: Comment here too

File: keck_vnc_launcher.py [hunk 3]
```diff
@@ -512,30 +515,35 @@ def open_ssh_tunnel(self, server, username, password, ssh_pkey, remote_port,
...
+            raise RuntimeError('subprocess failed to execute ssh')
+
+        checks = 50
```
Comment (joshwalawender) [inline keck_vnc_launcher.py:543]: Comment here as well please

**Review by joshwalawender — CHANGES_REQUESTED** [review]
Looks good as per our discussion earlier.  Would be good to get some explanatory comments in for future reference.

**Comment by klanclos** [comment]
I couldn't figure out how to successfully check out the branch on another machine, so I had to go back to my original checkout to make changes. I added several comments to the open_ssh_tunnel() routine.

**Review by joshwalawender — APPROVED** [review]

**Pull request merged**